### ISSUE TITLE
make TEST_TYPE=perf produces incomplete reports — op suite and Granite not measured on default path

### ISSUE THREAD

**Issue by Rafael-Sadykov** [torch-spyre/torch-spyre#4681]
### What happened?

`make TEST_TYPE=perf` invokes `spyre-perf-suite` on its default `hf_adapters`
path, which called only `run_hf_adapters()` and never iterated the model registry.
As a result, perf reports contained no Granite sections —
the run appeared complete but all models coverage was silently missing.

### What did you expect to happen?

We need to deploy the full list of spyre-perf-suite benchmarking,
Running `make TEST_TYPE=perf` should produce a report containing:

- matmul
- pointwise_add
- softmax
- rms_norm
- layer_norm
- attention_gqa
- sdpa
- mlp
- experimental.flash_attn
- granite-3.3-8b-instruct_bs1_pl512_prefill_only_full_model
- granite-3.3-8b-instruct_bs1_pl512_prefill_only_one_block
- granite-3.3-8b-instruct_bs4_pl128_decode_only_full_model
- granite-3.3-8b-instruct_bs4_pl128_decode_only_one_block
- granite-3.3-8b-instruct_bs4_pl128_generate_full_model
- granite-3.3-8b-instruct_bs4_pl128_generate_one_block

This is the full coverage the report-validation gate
(tests/test_report_validation.py) requires.

### How can we reproduce it?

On any torch-spyre commit prior to spyre-perf-suite PR # 191 being available
in the environment:

1. Run the perf target:

       make TEST_TYPE=perf

   which internally invokes:

       spyre-perf-suite --no-experimental --stacks torch-spyre \
           --report "$(RESULTS_DIR)/report.txt"

2. Inspect the generated report:

       grep -c '^Op:' report.txt
       grep -c '^Granite Config:' report.txt

Both return 0. The report contains no op sections and no Granite sections.

Root cause (two bugs in spyre-perf-suite):

Bug 1 — run_benchmark.py:
  `--mode hf_adapters` (the default) called only `run_hf_adapters()`.
  Only `--mode fms` called `run_benchmarks()`, which iterates the op registry.
  The two branches were mutually exclusive.

Bug 2 — core/report_generator.py:
  Under `hf_mode=True`, `generate_all_reports()` emptied the op file lists,
  dropping every `Op:` section from `report_hf.txt` silently.

### Any environmental details we need to know?

- Stack: torch-spyre (`--stacks torch-spyre`)
- Invocation: `spyre-perf-suite --no-experimental --stacks torch-spyre`
- Affected path: default `--mode hf_adapters` (no explicit --mode needed to hit the bug)
- Hardware: Spyre/AIU card required for a full run; the report-generator bug
  (Bug 2) is reproducible offline with the torch-free test added in
  tests/test_hf_report_op_sections.py
- Not version-specific: present on main prior to spyre-perf-suite PR #191

### Anything else we need to know?

Fix: spyre-perf-suite PR #191 (`feat/hf-adapters-run-op-suite`, merged) resolved
both bugs:

1. The `hf_adapters` branch now calls `run_benchmarks(stacks=["torch-spyre"])`
   before `run_hf_adapters()`, so the op registry is always iterated.
2. `core/report_generator.py` now collects op files under `hf_mode=True`
   the same way the non-hf path does.
3. A new `--hf-only` flag preserves the old models-only behaviour for callers
   that need it.

The torch-spyre Makefile invocation is correct and unchanged:

    spyre-perf-suite --no-experimental --stacks torch-spyre \
        --report "$(RESULTS_DIR)/report.txt"

No torch-spyre change is required. This issue is filed for traceability.

Related:
- spyre-perf-suite PR #191: Run the op suite on the default hf_adapters mode (fix, merged)
- spyre-perf-suite PR #4580: make: measure the op suite plus Granite in TEST_TYPE=perf (closed — fix belonged in spyre-perf-suite)

Remaining open item (separate follow-up needed):
- Every section reports `pt_util% = 0.000`. Predates this bug; present in the
  committed `docs/report.txt` on main. Requires `SENPERFORMANCE=2` set before
  `import torch` and a name-match correction in the profiling path.

### Relevant log output

```Shell
$ grep -c '^Op:' report.txt
0
$ grep -c '^Granite Config:' report.txt
0
```

**Comment by Vikram123-byte** [comment]
No torch-spyre PR. make TEST_TYPE=perf already runs spyre-perf-suite --no-experimental --stacks torch-spyre (not --mode fms; #4580 stays closed).

Last night’s x86 prod row 13004445407288905594 had 19 ops / 0 Granite, which matches this bug. Suite fix is spyre-perf-suite 191 (merged 17 Sep). INCOMPLETE/torch-spyre v0.0.1 is a different provenance issue (150 / 200), not this ticket.

Leaving open until the next nightly wheel includes 191 and Spyre View shows the six granite-3.3-8b-instruct configs. Then we close.

**Comment by Rafael-Sadykov** [comment]
Closing as completed. perf-suite PRs are active in the latest runs, and the report/dashboard now displays full coverage including all ops and Granite HF-adapter configurations.

Note: The INCOMPLETE run badge in a dashboard is due to provenance/version SHA tracking (torch-spyre version 0.0.1) and is being handled separately.
